### PR TITLE
DRAFT -  BFD-786: Fix Tax Number Mapping in V1 

### PR DESCRIPTION
**Pull request by cbrunefearless**
### Change Details
V1 had a bug with tax number  where it had a system and value but nothing to tell it what that value corresponds to.. Below is an example.
`{
    "sequence" : 3,
    "provider" : {
      "identifier" : {
        "system" : "http://terminology.hl7.org/CodeSystem/v2-0203",
        "value" : "204299999"
      }
    },
    "responsible" : true,
    "role" : {
      "coding" : [ {
        "system" : "http://hl7.org/fhir/claimcareteamrole",
        "code" : "other",
        "display" : "Other"
      } ]
    }
  }, `

In order to make it backwards compatible, we are keeping the current mapping and mapping another one with a code system.  Example below:
`{
    "sequence" : 3,
    "provider" : {
      "identifier" : {
        "system" : "http://terminology.hl7.org/CodeSystem/v2-0203",
        "value" : "204299999"
      }
    },
    "responsible" : true,
    "role" : {
      "coding" : [ {
        "system" : "http://hl7.org/fhir/claimcareteamrole",
        "code" : "other",
        "display" : "Other"
      } ]
    }
  }, {
    "sequence" : 4,
    "provider" : {
      "identifier" : {
        "type" : {
          "coding" : [ {
            "system" : "http://terminology.hl7.org/CodeSystem/v2-0203",
            "code" : "TAX",
            "display" : "Tax ID number"
          } ]
        },
        "system" : "http://terminology.hl7.org/CodeSystem/v2-0203",
        "value" : "204299999"
      }
    },
    "responsible" : true,
    "role" : {
      "coding" : [ {
        "system" : "http://hl7.org/fhir/claimcareteamrole",
        "code" : "other",
        "display" : "Other"
      } ]
    }
  } ],`

### Acceptance Validation
Make sure the code looks correct and its correct in the fihr spec.

### Feedback Requested
Check the code looks correct for v1 and v2.  Integration tests have been added to andrews faulks branch.

### External References

- [BFD-786](https://jira.cms.gov/browse/BFD-786)

### Security Implications

- [ ] new software dependencies
N/A

N/A
- [ ] altered security controls

N/A
- [ ] new data stored or transmitted

N/A
- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/carin/C4BBPractitionerIdentifierType.java [hunk 1]
```diff
@@ -27,6 +28,8 @@ public String toCode() {
         return "UPIN";
       case PIN:
         return "PIN";
+      case TAX:
+        return "PIN";
```
Comment (mackec1) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/carin/C4BBPractitionerIdentifierType.java:32]: cut-past error? Should be "TAX" shouldn't it?

Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/carin/C4BBPractitionerIdentifierType.java]: done

**Review by mackec1 — APPROVED** [review]
LGTM

**Review by mackec1 — APPROVED** [review]

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/carin/C4BBClaimProfessionalAndNonClinicianCareTeamRole.java [hunk 2]
```diff
@@ -10,7 +10,8 @@
   SUPERVISOR,
   PERFORMING,
   PURCHASED_SERVICE,
-  REFERRING;
+  REFERRING,
+  OTHER;
```
Comment (cbrunefearless) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/carin/C4BBClaimProfessionalAndNonClinicianCareTeamRole.java:14]: In V1 it is other, should it be other operating in v2? http://hl7.org/fhir/us/carin-bb/STU1/CodeSystem-C4BBClaimCareTeamRole.html

**Comment by rtravitz** [comment]
Would it be fair to convert this to a draft PR until it's able to be in a place that it could be merged again? Adds extra safety beyond the title change.

**Pull request closed without merging**